packages/docusaurus-theme-common/src/utils/useThemeConfig.ts

```diff
@@ -104,6 +104,10 @@ export type TableOfContents = {
 export type ThemeConfig = {
   docs: {
     versionPersistence: DocsVersionPersistence;
```

Comment:
This API is, AFAIK, undocumented. Is it intended for usage by anyone? I initially thought about `docsSidebar` so there's some degree of parallel with `navbar` and `footer`, but because of the existence of this field, I eventually put it in `docs.sidebar`, with one extra level of nesting.

Reply:
We should likely document `versionPersistence`, implemented in https://github.com/facebook/docusaurus/pull/3543

Note some sites may want to minimize the usage of local storage. But it may have been me misinterpreting the GDPR rules and may not be that useful to provide an option for it 🤷‍♂️ 

---

Nesting `docs.sidebar` doesn't look like a bad idea anyway, still related to [preemptive pluralization](https://www.swyx.io/preemptive-pluralization) 🤪